### DISCUSSION TITLE
SFDX Project configuration YML support

### DISCUSSION THREAD

**Discussion by dieffrei** [forcedotcom/cli#2350] [Ideas]
**Is your feature request related to a problem? Please describe.**
sfdx-project.json file becomes very difficult to read when it is big, currently, we already have +50 packages and for the next year probably it is going to be around +100.
Large text files lead to merge conflicts, also is very difficult to understand multi-directory projects.

**What are you trying to do**
I would like to have more flexibility on sfdx-project.json files. E.g: group packages per domain, provide them some capabilities like comments, or even have more granular configurations (per package).

**Describe the solution you'd like**
YML support E.g (sfdx-project.yml) 

**Additional context**
Large sfdx-project.json files

[a comment by an automated account is not shown]

**Comment by yippie** [comment]
To add to this, when there are lots of packages with dependencies, needing to specify the entire dependency tree for every package is very verbose and cumbersome. Supporting cascading dependencies would be super helpful.

**Comment by AllanOricil** [comment]
I've created a proposal for this issue.

https://github.com/forcedotcom/cli/issues/1915

Just copy (get inspired by) what maven did with pom files. Each package directory will have its own `sfdx-package.json`. The sfdx-project.json will behave as if it was an aggregator pom when working with maven projects.

This change will ease the implementation of a sfdx reactor (like mvn reactor)

https://github.com/forcedotcom/cli/issues/1950

**Comment by mshanemc** [comment]
json not allowing comments I totally understand.  Can you elaborate on how yml solves those problems better than json?
